### PR TITLE
[common] fix skipping download of file existence in some cases

### PR DESCRIPTION
**Pull request by junzh0u**
Current logic of skipping download when file already exists have some bugs when working together with ffmpeg-merge.

TL;DR: When merging multi parts using ffmpeg, we change the ext name of the output file. Thus messed up with the current skipping download mechanism.

Considering a video with title "TITLE", it have multiple parts, all with ext ".flv".
We will look for "TITLE.flv" first. If it doesn't exist, we will download all parts as "TITLE[00].flv", "TITLE[01].flv"... etc.
Then if we don't have ffmpeg installed, we merged them into "TITLE.flv".

So far so good.
But problem happens if we have ffmpeg installed. We will merge all parts to "TITLE.mp4". That means when we try to download the same video again, it will not be able to find any file named "TITLE.flv". Thus it will download again.

This diff solves this issue.

[Review on Reviewable](https://reviewable.io/reviews/soimort/you-get/646)

### CODE REVIEW

**Comment by soimort** [comment]
Sounds reasonable. Thanks for the contribution!
